### PR TITLE
only apply view extension if azimuth and zenith are not NaN

### PR DESCRIPTION
**Pull request by philvarner**
- https://github.com/stactools-packages/sentinel2/issues/101

### CODE REVIEW

**Review by pjhartzell — APPROVED** [review]

**Comment by philvarner** [comment]
looks like there are some assets that have NaN for those fields, so there must be multiple places where the fields are set.

File: src/stactools/sentinel2/stac.py [hunk 1]
```diff
@@ -364,8 +373,17 @@ def set_asset_properties(_asset: pystac.Asset, _band_gsd: Optional[int] = None):
         viewing_angle = viewing_angles[band_id]
         # We can't use the ViewExtension here until
         # https://github.com/stac-utils/pystac/issues/793 is fixed
-        asset.extra_fields["view:azimuth"] = viewing_angle.azimuth
-        asset.extra_fields["view:incidence_angle"] = viewing_angle.zenith
+        if not math.isnan(viewing_angle.azimuth):
```
Comment (philvarner) [inline src/stactools/sentinel2/stac.py:376]: I filed https://github.com/stac-extensions/view/issues/4 to discuss whether these fields should be allowed by the spec on Assets

**Review by pjhartzell — APPROVED** [review]
You've run across instances where the sun_azimuth and sun_zenith are NaNs, but asset azimuths and/or zeniths exist? The changes make sense in terms of handling that scenario.

**Comment by philvarner** [comment]
> You've run across instances where the sun_azimuth and sun_zenith are NaNs, but asset azimuths and/or zeniths exist? The changes make sense in terms of handling that scenario.

The filer's example has values for sun_azimuth and sun_zenith, but some of the assets have a & z values and some have NaN. The test example was contrived and the code written defensively to handle all these cases, even though I don't know if the sun_azimuth and sun_zenith are ever NaN in practice.

**Pull request merged**